File: wx-login-service/src/main/java/com/example/wxlogin/entity/CompanyInfo.java
package com.example.wxlogin.entity;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.Data;

import java.time.LocalDateTime;

/**
 * 企业信息实体
 */
@Data
@TableName("company_info")
public class CompanyInfo {

    @TableId(type = IdType.AUTO)
    private Long id;

    /** 关联用户openid */
    private String openid;

    /** 企业名称 */
    private String name;

    /** 统一社会信用代码 */
    private String creditCode;

    /** 联系人 */
    private String contact;

    /** 联系电话 */
    private String phone;

    /** 企业地址 */
    private String address;

    /** 备注 */
    private String remark;

    /** 显卡数量范围 */
    private String gpuCount;

    /** 服务需求(JSON数组) */
    private String services;

    /** 状态: pending/approved/rejected */
    private String status;

    /** 是否已使用免费上门检测 */
    private Integer hasUsedFreeService;

    /** 提交时间 */
    private LocalDateTime submitTime;

    /** 更新时间 */
    private LocalDateTime updatedAt;
}

File: gpu_diagnostic_app/lib/core/models/company_info.dart
/// 企业信息模型
class CompanyInfo {
  final int? id;
  final String openid;
  final String name;
  final String? creditCode;
  final String contact;
  final String phone;
  final String address;
  final String? remark;
  final String? gpuCount;
  final List<String>? services;
  final String status;
  final bool hasUsedFreeService;

  CompanyInfo({
    this.id,
    required this.openid,
    required this.name,
    this.creditCode,
    required this.contact,
    required this.phone,
    required this.address,
    this.remark,
    this.gpuCount,
    this.services,
    this.status = 'pending',
    this.hasUsedFreeService = false,
  });

  factory CompanyInfo.fromJson(Map<String, dynamic> json) {
    return CompanyInfo(
      id: json['id'],
      openid: json['openid'] ?? '',
      name: json['name'] ?? '',
      creditCode: json['creditCode'] ?? json['credit_code'],
      contact: json['contact'] ?? '',
      phone: json['phone'] ?? '',
      address: json['address'] ?? '',
      remark: json['remark'],
      gpuCount: json['gpuCount'] ?? json['gpu_count'],
      services: json['services'] != null
          ? List<String>.from(json['services'])
          : null,
      status: json['status'] ?? 'pending',
      hasUsedFreeService:
          json['hasUsedFreeService'] == 1 || json['has_used_free_service'] == 1,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'openid': openid,
      'name': name,
      'creditCode': creditCode,
      'contact': contact,
      'phone': phone,
      'address': address,
      'remark': remark,
      'gpuCount': gpuCount,
      'services': services,
    };
  }
}
